client/src/services/site.js

```diff
@@ -1,6 +1,19 @@
 import store from 'store';
 import { API_URL } from '../constants';
 
+export const fetchSitePhases = async (siteId) => {
+  const response = await fetch(`${API_URL}/api/v1/phase-allocation/${siteId}`, {
+    headers: { Authorization: `Bearer ${store.get('TOKEN')}` },
+    method: 'GET',
+  });
+  let phases = [];
+  if (response.ok) {
+    phases = await response.json();
+    phases = phases.data;
```

Comment:
Good point, I tidied it up a bit. I'd say it's still flawed- if it either fails to fetch the site's participants (fetchSiteRows is the next function in this file) or fails to fetch phases, in both cases it returns [] and doesn't toast or otherwise communicate to the user.